bundles/org.eclipse.swt/Eclipse SWT/cocoa/org/eclipse/swt/widgets/Display.java

```diff
@@ -613,6 +615,16 @@ public void asyncExec (Runnable runnable) {
 	}
 }
 
+@Override
+public void execute(Runnable command) {
```

Comment:
Please add comprehensive javadoc. The one provided by super class is misleading, because **any** execution in SWT happens on UI thread and nowhere else.
Please also provide some simple example of the possible use of this method *in the javadoc* here, so developers have an idea what it is good for, with a **bold** warning that no guarantees will be given if this returns synchronously or not, because it depends on the thread in which the current code is running.

Reply:
I have added the javadoc now, I hope it is suitable.

Reply:
@iloveeclipse are you satisfied with the changed javadoc?

Reply:
@iloveeclipse could you check if the javadoc fit now or like to recommend more changes? I would otherwise assume this as resolved?

Reply:
Looks better now, just few smaller issues.